src/turbomind/models/llama/LlamaBatch.cc

```diff
@@ -1050,17 +1052,48 @@ void LlamaBatch<T>::InitializeSampling(const GenerationState& g)
             FT_CHECK(shape[0] == 1);
             shape[0]                = batch_size;
             const int size_in_bytes = ref.sizeBytes();
-            memset(h_ptr, 0, size_in_bytes * batch_size);
+
+            int max_list_length = 0;
+            if (name == "bad_words_list" || name == "stop_words_list") {
+                for (int i = 0; i < batch_size; ++i) {
+                    if (state_->requests[i]->inputs[rank_].isExist(name)) {
+                        Tensor& src = state_->requests[i]->inputs[rank_].at(name);
+                        FT_CHECK(src.shape.size() == 3 && src.shape[1] == 2 && src.shape[2] <= kMaxStopBadWordsLen);
+                        max_list_length = std::max(max_list_length, (int)src.shape[2]);
+                    }
+                }
+                std::fill_n((int*)h_ptr, batch_size * 2 * max_list_length, -1);
```

Comment:
`std::fill_n((int*)h_ptr, batch_size * 2 * max_list_length, -1);` should be
`std::fill_n((int*)h_ptr, batch_size *  size_in_bytes * 2 * max_list_length, -1);`